## Add the Target Variable
Now that the data from r/Democrats and r/Conservative have been acquired using the API.  The target variable must be added.  The study is not trying to predict a particular group so the assignment of the values to the target variable is arbitrary.  In this case r/Democrat will be assigned 0 and the r/Conservative will be assigned 1. 

In [48]:
import pandas as pd

In [49]:
dems = pd.read_csv('./data/democratic.csv')
reps = pd.read_csv('./data/republican.csv')


In [50]:
dems.head(1)

approved_at_utc  approved_by  archived         author author_cakeday  \
0              NaN          NaN     False  backpackwayne            NaN   

   author_flair_background_color  author_flair_css_class  \
0                            NaN                     NaN   

  author_flair_richtext  author_flair_template_id  author_flair_text ...   \
0                    []                       NaN                NaN ...    

  thumbnail_height thumbnail_width  \
0              NaN             NaN   

                                               title  ups  \
0  THE TIME FOR UNITY IS NOW - A Progressive and ...   79   

                                                 url  user_reports  \
0  https://www.reddit.com/r/democrats/comments/95...            []   

   view_count  visited  whitelist_status  wls  
0         NaN    False           all_ads    6  

[1 rows x 97 columns]

In [51]:
dems.shape


(876, 97)

In [52]:
reps.head(3)

approved_at_utc  approved_by  archived           author author_cakeday  \
0              NaN          NaN     False         tehForce            NaN   
1              NaN          NaN     False  thatrightwinger            NaN   
2              NaN          NaN     False        chabanais            NaN   

   author_flair_background_color author_flair_css_class  \
0                            NaN                    NaN   
1                            NaN              Far-Right   
2                            NaN                   Bold   

                               author_flair_richtext  \
0  [{'e': 'text', 't': 'Liberal Education != Purp...   
1          [{'e': 'text', 't': "Don't Tread on Me"}]   
2         [{'e': 'text', 't': 'Stronger than derp'}]   

               author_flair_template_id                 author_flair_text  \
0                                   NaN  Liberal Education != Purple Hair   
1  9b86186a-8b38-11e1-8f58-12313d2c1af1                 Don't Tread on Me   
2                                   NaN                Stronger than derp   

  ...  thumbnail_height thumbnail_width  \
0 ...               NaN             NaN   
1 ...             105.0           140.0   
2 ...             140.0           140.0   

                                               title   ups  \
0    Michael Oakeshott - this week's sidebar tribute    41   
1  Watch Live: Judge Brett Kavanaugh's confirmati...    91   
2                      Ben Shapiro lays them away...  1426   

                                                 url  user_reports  \
0  https://www.reddit.com/r/Conservative/comments...            []   
1        https://www.youtube.com/watch?v=h8il9dAhA00            []   
2                https://i.redd.it/i3xkb1sgb8k11.jpg            []   

   view_count  visited  whitelist_status  wls  
0         NaN    False           all_ads    6  
1         NaN    False           all_ads    6  
2         NaN    False           all_ads    6  

[3 rows x 97 columns]

In [53]:
reps.shape

(861, 97)

Both data sets have 97 variables.  r/democrat has 876 entries and the r/Conservative has 861 entries.  

### Adding target value

In [54]:
dems['target'] = 0
dems['target'].head()

0    0
1    0
2    0
3    0
4    0
Name: target, dtype: int64

In [55]:
reps['target'] = 1
reps['target'].head()

0    1
1    1
2    1
3    1
4    1
Name: target, dtype: int64

### Merging the dataframes

Each data from should now have 98 features and they can be merged together for analysis

In [56]:
frames = [dems, reps]
comb = pd.concat(frames, axis = 0)

In [57]:
comb.shape

(1737, 98)

It looks as if the data frames combined correctly.  Still have 94 features and have 1737 cases which is 861 + 876

Next the excess variables will be striped and the DataFrame will be save to a csv to be read into an analysis notebook.  The vectorization will be done with the analysis, since the parameters of the vectorization will be altered to maximize both the prediction and the explanatory power of the model.

In [58]:
comb.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1737 entries, 0 to 860
Data columns (total 98 columns):
approved_at_utc                  0 non-null float64
approved_by                      0 non-null float64
archived                         1737 non-null bool
author                           1737 non-null object
author_cakeday                   2 non-null object
author_flair_background_color    0 non-null float64
author_flair_css_class           261 non-null object
author_flair_richtext            1735 non-null object
author_flair_template_id         175 non-null object
author_flair_text                552 non-null object
author_flair_text_color          554 non-null object
author_flair_type                1735 non-null object
author_fullname                  1735 non-null object
banned_at_utc                    0 non-null float64
banned_by                        0 non-null float64
can_gild                         1737 non-null bool
can_mod_post                     1737 non-null bool

In [59]:
comb.title.head(4)


0    THE TIME FOR UNITY IS NOW - A Progressive and ...
1                                            Shameful.
2    Kavanaugh Hearing Erupts in Chaos as Dems Dema...
3    Democrats are driving home that Republicans ke...
Name: title, dtype: object

The title will contain information that will be helpful in sorting out issues and words that identify Democratic posts v. Conservative posts

In [60]:
comb.url.value_counts()

http://www.foxnews.com/us/2018/08/29/judge-dismisses-all-charges-against-3-extremist-muslim-new-mexico-compound-suspects.html                                                                                 4
https://babylonbee.com/news/democrats-demand-kavanaugh-submit-to-dna-test-to-prove-hes-not-actually-hitler/                                                                                                   3
http://dailycaller.com/2018/08/31/cnn-list-bungled-reporting-fake-news/                                                                                                                                       2
https://dailycaller.com/2018/09/04/zina-bash-kavanaugh-hearing-trump-video/                                                                                                                                   2
https://www.washingtonexaminer.com/washington-secrets/us-accounts-for-just-1-of-mass-shootings-not-media-hyped-31                                                       